server/controller/challenge.js

```diff
@@ -0,0 +1,189 @@
+const { challengeDB } = require("../Model/index");
+const Joi = require("@hapi/joi");
+
+exports.getExpertiseChallenges = async (req, res) => {
+  const expertise_id = req.params.expertise_id;
+
+  try {
+    let challenges = await challengeDB.find({ expertise_id });
+    res.status(200).json({
+      msg: challenges,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      error: "Server Error!",
+    });
+  }
+};
+exports.addChallenge = async (req, res) => {
+  const { sub } = req.user;
+  const {
+    topic,
+    expertise,
+    difficulty,
+    description,
+    repo_url,
+    assets,
+    cover_photo,
+  } = req.body;
+
+  const schema = Joi.object().keys({
+    topic: Joi.string().required(),
+    expertise: Joi.string().required(),
+    difficulty: Joi.number().min(1).max(3).required(),
+    description: Joi.string().required(),
+    repo_url: Joi.string(),
+    assets: Joi.string(),
+    cover_photo: Joi.string(),
+  });
+
+  // Schema Validation
+  try {
+    await schema.validateAsync({
+      topic,
+      expertise,
+      difficulty,
+      description,
+      repo_url,
+      assets,
+      cover_photo,
+    });
+  } catch (error) {
+    return res.status(400).json({
+      error: error.message,
+    });
+  }
+
+  try {
+    // Find an Expertise with topic [expertise] and auth0_ref [sub]
+    const expertise = expertise.findOne({ topic: expertise, auth0_ref: sub });
```

Comment:
it is better to use camelcase `auth0_ref` to `auth0Ref`

Reply:
Forgot to fix this

Reply:
Updated!